Distante algoritmi optimizare rute

Anna1070 opened the pull request.

Anna1070 committed:
Created a new branch and a controller

Anna1070 committed:
Task 2: Managed to build the distances matrix

Added the functionality to build the distances matrix and writing it down in an excel file for an easier observation and also in a txt file.

Anna1070 committed:
Disabled the distance matrix button

Anna1070 committed:
Create Traseu_2_SB_77_ULB.xlsx

paulaeneotescu committed:
Insert new entries for Route 2

Now, there are 332 locations.
Also, I made some changes on frontend.

paulaeneotescu committed:
Task3: Optimized Route

I added the "ClosedXML" package to read the distances directly from the .xlsx file. I used the "Leaflet Routing Machine" (free version) to display the route on the streets.

pdany1116 commented on Proiect_final/Colectare_pubela/Colectare_pubela/Controllers/HartaController.cs:
```diff
@@ -22,11 +22,92 @@ public IActionResult AfiseazaHarta()
+                        matrice[i, j] = distanta;
+                    } else
+                    {
+                        matrice[i, j] = double.MaxValue;
```
De ce setam max value daca avem o eroare de parsare a matricii?

paulaeneotescu replied:
O setez cu maxim, deoarece daca cumva valoarea din matrice e invalida sau e un caz care nu e ok, acesta se va evita. Din ce am vazut, in legatura cu algoritmii TSP, ar putea exista cazul de distanta "infinita", iar double.MaxValue poate marca agest caz, si va fi ignorat.  Cred ca se putea seta si cu null, dar mai trebuiau adaugate modificari la unele lunii de cod.

pdany1116 commented on Proiect_final/Colectare_pubela/Colectare_pubela/Views/Shared/_Layout.cshtml:
```diff
@@ -44,6 +45,7 @@
     <script src="~/lib/jquery/dist/jquery.min.js"></script>
     <script src="~/lib/bootstrap/dist/js/bootstrap.bundle.min.js"></script>
     <script src="https://unpkg.com/leaflet@1.9.4/dist/leaflet.js"></script>
+    <script src="https://unpkg.com/leaflet-routing-machine@latest/dist/leaflet-routing-machine.js"></script>
```
Felicitari! Ador ca ati folosit aceasta librarie sa aratai rutele pe bune! 🚀

pdany1116 reviewed (commented):
Buna treaba! As dori sa vad si diferenta de km intre ruta initiala si cea optimizata, altfel nu vom stii niciodata daca optimizarea e intr-adevar optimizare :)

Anna1070 committed:
Task 4 and Task 5

I calculated the distances for the default route and the optimized route. Calculated the estimated time for the optimized route to be completed in.

pdany1116 commented on Proiect_final/Colectare_pubela/Colectare_pubela/Controllers/HartaController.cs:
```diff
@@ -31,8 +31,23 @@ public IActionResult AfiseazaHarta()
+            var rutaNormala = Enumerable.Range(0, matriceaDistantelor.GetLength(0)).ToList();
             var rutaOptimizata = GasesteRutaOptimizata(matriceaDistantelor);
+
+            double distantaNormala = CalculeazaDistantaTotala(rutaNormala, matriceaDistantelor);
+            double distantaOptimizata = CalculeazaDistantaTotala(rutaOptimizata, matriceaDistantelor);
```
Distantele nu sunt calculate corect, distanta intre start si primul punct nu este luata in considerare, nici distanta dintre ultimul punct si stop

Anna1070 committed:
Added the starting point and the end point

Anna1070 committed:
Update HartaController.cs

Anna1070 committed:
Problems solved
